### PR TITLE
i18n(es): update `middleware`

### PR DESCRIPTION
**Pull request by dreyfus92**
#### Description (required)

updates `middleware`

#### Related issues & labels (optional)

- Closes # N/A
- Suggested label: i18n

### CODE REVIEW

File: src/content/docs/es/guides/middleware.mdx [hunk 1]
```diff
@@ -198,6 +199,74 @@ respuesta de saludo
...
+<p><Since v="4.13.0" /></p>
+
+El `APIContext` expone un método llamado `rewrite()` que funciona de la misma manera que [Astro.rewrite](/es/guides/routing/#rewrites).
+
+Usa `context.rewrite()` dentro del middleware para mostrar el contenido de una página diferente sin [redirigir](/es/guides/routing/#redirecciones-dinámicas) a tu visitante a una nueva página. Esto desencadenará una nueva fase de renderizado, lo que hará que cualquier middleware se vuelva a ejecutar.
+
+```js title="src/middleware.js"
+import { isLoggedIn } from "~/auth.js"
+export function onRequest (context, next) {
+  if (!isLoggedIn(context) {
```
Comment (liruifengv) [inline src/content/docs/es/guides/middleware.mdx:214]:
```suggestion
  if (!isLoggedIn(context)) {
```

I fixed this in https://github.com/withastro/docs/pull/8994

File: src/content/docs/es/guides/middleware.mdx [hunk 2]
```diff
@@ -198,6 +199,74 @@ respuesta de saludo
...
+  }
+
+  return next();
+};
+```
+Puedes pasarle a la función `next()` un parámetro de ruta de URL opcional para reescribir la `Request` actual sin volver a desencadenar una nueva fase de renderizado. La ubicación de la ruta de reescritura se puede proporcionar como una cadena, URL o `Request`:
+```js title="src/middleware.js"
+import { isLoggedIn } from "~/auth.js"
+export function onRequest (context, next) {
+  if (!isLoggedIn(context) {
```
Comment (liruifengv) [inline src/content/docs/es/guides/middleware.mdx:232]:
```suggestion
  if (!isLoggedIn(context)) {
```

**Comment by dreyfus92** [comment]
Thank you for the review @liruifengv 🫡

**Review by yanthomasdev — COMMENTED** [review]
Here we have a similar issue where we have moved a few things, so we'd need to update [`routing.mdx`](https://github.com/withastro/docs/commits/main/src/content/docs/en/guides/routing.mdx?since=2024-03-18T13:52:22.000Z) to have this one merged, could you take a look and update when you have the chance? 🙌

**Comment by yanthomasdev** [comment]
I'm closing this PR since it got stale and outdated, feel free to open it again later!

**Pull request closed without merging**